### PR TITLE
Document compatibility between declarative and procedural macro tokens

### PR DESCRIPTION
**Pull request by petrochenkov**
This PR assumes https://github.com/rust-lang/rust/pull/92472 is merged.

### CODE REVIEW

File: src/procedural-macros.md [hunk 1]
```diff
@@ -271,12 +271,72 @@ fn invoke4() {}
...
+
+When passed to a proc-macro
+- All multi-character operators are broken into single characters.
+- Lifetimes are broken into a `'` character and an identifier.
+- All metavariable substitutions are represented as their underlying token
+  streams.
+    - Such token streams may be wrapped into delimited groups ([`Group`]) with
+      implicit delimiters ([`Delimiter::None`]) when it's necessary for
+      preserving parsing priorities.
+    - `tt` and `ident` substitutions are never wrapped into such groups and
```
Comment (petrochenkov) [inline src/procedural-macros.md:316]: The `ident` part here requires https://github.com/rust-lang/rust/pull/92472.

File: src/procedural-macros.md [hunk 2]
```diff
@@ -271,12 +271,72 @@ fn invoke4() {}
...
+      always represented as their underlying token trees.
+
+When passed from a proc macro
```
Comment (dtolnay) [inline src/procedural-macros.md:319]: I suggest "When emitted from a proc macro" or "When returned by a proc macro". It's passing in a sense, but this isn't the clearest way to say that.

**Review by dtolnay — CHANGES_REQUESTED** [review]

**Comment by petrochenkov** [comment]
I've addressed review comments.
https://github.com/rust-lang/rust/pull/92472 has also landed.

**Comment by petrochenkov** [comment]
@rustbot ready
(Is the bot working on this repo?)

**Review by ehuss — COMMENTED** [review]
Thanks!

File: src/procedural-macros.md [hunk 3]
```diff
@@ -271,12 +271,72 @@ fn invoke4() {}
...
+- Lifetimes (`'ident`)
+- Metavariable substitutions in `macro_rules` (e.g. `$my_expr` in
+  `macro_rules! mac { ($my_expr: expr) => { $my_expr } }` after the `mac`'s
+  expansion, which will be considered a single token tree regardless of the
+  passed expression)
+
+Token trees in procedural macros are defined as
+- Delimited groups (`(...)`, `{...}`, etc)
+- All punctuation characters used in operators supported by the language (`+`,
+  but not `+=`), and also the single quote `'` character
```
[the remark this answers is not in the document]
Comment (petrochenkov) [reply src/procedural-macros.md]: I languages defined by proc macros `'` is an independent token, it doesn't necessarily have to be joined with identifiers to form lifetimes.

**Comment by petrochenkov** [comment]
I've added the changes from https://github.com/rust-lang/reference/pull/1169#discussion_r839962540 and https://github.com/rust-lang/reference/pull/1169#discussion_r839968049 in a slightly reworded form.

**Review by ehuss — APPROVED** [review]
Thanks!

**Pull request merged**